webapp/src/ts/services/enketo.service.ts

```diff
@@ -901,16 +886,38 @@ interface XmlFormContext {
   instanceData: null|string|Record<string, any>; // String for report forms, Record<> for contact forms.
   titleKey?: string;
   isFormInModal?: boolean;
-  contactSummary: Record<string, any>;
+  contactSummary?: Record<string, any>;
 }
 
-export interface EnketoFormContext {
+export class EnketoFormContext {
   selector: string;
   formDoc: Record<string, any>;
-  instanceData: null|string|Record<string, any>; // String for report forms, Record<> for contact forms.
+  type: 'contact'|'report'|'task'|'training-card';
+  editing: boolean;
+  instanceData: null|string|Record<string, any>;
   editedListener: () => void;
   valuechangeListener: () => void;
   titleKey?: string;
   isFormInModal?: boolean;
   userContact?: Record<string, any>;
+  contactSummary? :Record<string, any>;
+
+  constructor(selector?, type?, formDoc?, data?) {
+    this.selector = selector;
+    this.type = type;
+    this.formDoc = formDoc;
+    this.instanceData = data;
+  }
+
+  evaluateExpression() {
+    if (this.type === 'report' && this.editing) {
```

Comment:
Wouldn't this also apply when editing contacts?

Reply:
I'm worried that that might be desired behavior?

Reply:
Coming back to this: the contact add and edit forms are different, so the expression can be different between them. I'm tempted to think that we should allow blocking contact edits depending on expression.